### PR TITLE
Mem and led

### PR DESCRIPTION
**Pull request by asadate**
Added EEPROM, Flash memory, LED specifications and, examples for these three components and the BJT.

### CODE REVIEW

File: part-spec/led.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
+{
+    "$id": "https://github.com/chromeos/digital-datasheets/blob/main/part-spec/semiconductor.json",
```
Comment (rsnancollas) [inline part-spec/led.json]: should this file be in semiconductor.json instead of led.json?

File: part-spec/led.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                },
+                "iv": {
+                    "description": "LED luminous intensity",
+                    "$comment": "units of candela (cd). Measurement conditions include LED forward current, if",
+                    "$ref": "definitions.json#/conditionalProperty"
+                },
+                "peakWavelength": {
+                    "description": "light spectrum output value emitted by an LED at highest wavelength",
+                    "$comment": "units of nanometer(nm). Measurement conditions include LED forward current, if",
+                    "$ref": "definitions.json#/conditionaProperty"
```
Comment (rsnancollas) [inline part-spec/led.json]: conditionalProperty

File: part-spec/led.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                },
+                "dominantWavelength": {
+                    "description": " dominant wavelength an LED emits the majority of the time",
+                    "$comment": "units of nanometer(nm). Measurement conditions include LED forward current, if",
+                    "$ref": "definitions.json#/conditionalProperty"
+                },
+                "angleHalfIntensity": {
+                    "description": " angle at which LED intensity falls to 50% of its maximum value",
+                    "$comment": "units of degrees (celsius)",
```
Comment (rsnancollas) [inline part-spec/led.json]: probably not degrees Celsius, correct?

File: part-spec/storage_memory.json [hunk 4]
```diff
@@ -176,6 +226,186 @@
...
+                    "description": "number of rows in the eeprom",
+                    "type": "number"
+                },
+                "bitsPerWords": {
+                    "description": "number of columns in the eeprom",
+                    "type": "number"
+                },
+                "bootBlockSize": {
+                    "description": "size of the eeprom boot block",
+                    "comment": "units of bytes (B)",
```
Comment (rsnancollas) [inline part-spec/storage_memory.json]: why is the bootBlockSize in units of bytes, but the capacity in units of bits? Seems like they should be consistent OR the thing that's smaller should be in units of bits and the larger thing should be in units of bytes.

File: part-spec/storage_memory.json [hunk 5]
```diff
@@ -176,6 +226,186 @@
...
+                        "i2c",
+                        "serial 1-wire"
+                    ]
+                },
+                "clockFrequency": {
+                    "description": "eeprom clock frequency",
+                    "comment": "units of Hertz (Hz)",
+                    "$ref": "definitions.json#/unit"
+                },
+                "accessTime": {
```
Comment (rsnancollas) [inline part-spec/storage_memory.json:291]: I assume this is a standard definition, but is this from like: i2c write to data out?

File: part-spec/storage_memory.json [hunk 6]
```diff
@@ -176,6 +226,186 @@
...
+                    "$ref": "definitions.json#/componentID"
+                },
+                "capacity": {
+                    "description": "capacity/density of flash memory",
+                    "comment": "units of bits",
+                    "$ref": "definitions.json#/unit"
+                },
+                "pageSize": {
+                    "description": "page size of flash memory",
+                    "comment": "units of bytes (B)",
```
Comment (rsnancollas) [inline part-spec/storage_memory.json]: same bits vs bytes comment. Why is the page size in bytes, but the capacity in bits?

File: part-spec/storage_memory.json [hunk 7]
```diff
@@ -176,6 +226,186 @@
...
+                    "$ref": "definitions.json#/unit"
+                },
+                "blockSize": {
+                    "description": "block size of flash memory",
+                    "comment": "units of bytes (B)",
+                    "$ref": "definitions.json#/unit"
+                },
+                "bootBlockSize": {
+                    "description": "size of the fash memory boot block",
```
Comment (rsnancollas) [inline part-spec/storage_memory.json]: flash (not fash)

File: part-spec/storage_memory.json [hunk 8]
```diff
@@ -176,6 +226,186 @@
...
+                    "comment": "units of bytes (B)",
+                    "$ref": "definitions.json#/unit"
+                },
+                "interface": {
+                    "description": "interface of eeprom to host",
```
Comment (rsnancollas) [inline part-spec/storage_memory.json]: interface of flash memory

File: support-docs/examples.md [hunk 9]
```diff
@@ -2025,10 +2025,459 @@ Converter Example - Limited pins
...
+                "unitText": "mA",
+                "unitFactor": 0.001,
+                "valueDefined": true
+            }
+        }
+    },
+    "angleHalfIntensity": {
+        "typValue": 25,
+        "siUnit": "celsius",
+        "unitText": "degrees celsius",
```
Comment (rsnancollas) [inline support-docs/examples.md]: just degrees, right?

**Review by rsnancollas — APPROVED** [review]
I'm going to go ahead and approve this since the changes are fairly small. Please take a look at the feedback and make changes as necessary before submitting :)

**Review by rsnancollas — APPROVED** [review]
I'm going to go ahead and approve this since the changes are fairly small. Please take a look at the feedback and make changes as necessary before submitting :)

**Review by rsnancollas — APPROVED** [review]
I'm going to go ahead and approve this since the changes are fairly small. Please take a look at the feedback and make changes as necessary before submitting :)

File: part-spec/led.json (hunk 1 above)
Comment (asadate) [reply part-spec/led.json]: changed to led.json.  LED can be in either file. I don't have a strong preference.

File: part-spec/led.json (hunk 3 above)
Comment (asadate) [reply part-spec/led.json]: correct. not celcius

File: part-spec/storage_memory.json (hunk 5 above)
Comment (asadate) [reply part-spec/storage_memory.json:291]: This is from the interface supported by the component.

File: part-spec/storage_memory.json (hunk 6 above)
Comment (asadate) [reply part-spec/storage_memory.json]: changed to bits

File: support-docs/examples.md (hunk 9 above)
Comment (asadate) [reply support-docs/examples.md]: right

**Comment by asadate** [comment]
Added additional Memory and LED specifications

**Pull request merged**